app/src/androidTest/java/com/example/wellfed/StoredIngredientDBTest.java

```diff
@@ -0,0 +1,188 @@
+import com.example.wellfed.ingredient.StoredIngredientDB;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Date;
+
+@RunWith(AndroidJUnit4.class)
+public class StoredIngredientDBTest {
+    StoredIngredientDB storedIngredientDB;
+    @Before
+    public void before() {
+         storedIngredientDB = new StoredIngredientDB();
+    }
+
+    /**
+     * This method tests the add functionality and get functionality of db,
+     * with a complete ingredient.
+     * @throws InterruptedException
+     */
+    @Test
+    public void testAddFull() throws InterruptedException {
+        StoredIngredient storedIngredient = new StoredIngredient("Broccoli");
+        storedIngredient.setCategory("Vegetable");
+        Date bestBefore = new Date(2022, 10, 1);
+        storedIngredient.setBestBefore(bestBefore);
+        storedIngredient.setLocation("Fridge");
+        storedIngredient.setAmount(5.0f);
+        storedIngredient.setUnit("kg");
+
+        // testing whether it was what was inserted into db
+        String id = storedIngredientDB.addStoredIngredient(storedIngredient);
+        StoredIngredient resultIngredient = storedIngredientDB.getStoredIngredient(id);
+        assertEquals("Broccoli", resultIngredient.getDescription());
+        assertEquals("Vegetable", resultIngredient.getCategory());
+        assertEquals(bestBefore, resultIngredient.getBestBefore());
+        assertEquals("Fridge", resultIngredient.getLocation());
+        assertEquals(5.0f, resultIngredient.getAmount(), 0.01);
+        assertEquals("kg", resultIngredient.getUnit());
+
+
+        // removing it afterward
+        storedIngredientDB.removeFromIngredients(id);
+    }
+
+    /**
+     * This method tests the add and get functionality, when fields are blank.
+     * @throws InterruptedException
+     */
+    @Test
+    public void testAddMissingFields() throws InterruptedException {
+        StoredIngredient storedIngredient = new StoredIngredient("Broccoli");
+
+        // testing whether it was what was inserted into db
+        String id = storedIngredientDB.addStoredIngredient(storedIngredient);
+        StoredIngredient resultIngredient = storedIngredientDB.getStoredIngredient(id);
+        assertEquals("Broccoli", resultIngredient.getDescription());
+        assertNull(resultIngredient.getCategory());
+        assertNull(resultIngredient.getBestBefore());
+        assertNull(resultIngredient.getLocation());
+        assertEquals(0.0f, resultIngredient.getAmount(), 0.01);
+        assertNull(resultIngredient.getUnit());
+
+        // removing it afterward
+        storedIngredientDB.removeFromIngredients(id);
+    }
+
+    /**
+     * This method tests deleting a non-existing ingredient.
+     */
+    @Test
+    public void deleteNonExistingIngredient() {
+        // attempting to remove non-existing ingredient from db
+        // this test will succeed if no error is thrown
```

Comment:
Shouldn't it throw an error if the ingredient is not found? Why this way?

Reply:
My logic was that a non-existing ingredient being deleted should do nothing - for example, what if there was a case in the UI where a duplicate delete was accidentally called, by some quirk of a button press/a UI element not being hidden quickly enough, or something like that. It wouldn't hurt to have nothing deleted, and the error handling of this one case would be more issue prone than simply doing nothing.

Reply:
I guess that is okay, hopefully our other code never creates a case where this is possible.